src/Magento/FunctionalTestingFramework/StaticCheck/ActionGroupArgumentsCheck.php

```diff
@@ -0,0 +1,209 @@
+     * Array containing all errors found after running the execute() function.
+     * @var array
+     */
+    private $errors = [];
+
+    /**
+     * String representing the output summary found after running the execute() function.
+     * @var string
+     */
+    private $output;
+
+    /**
+     * Checks unused arguments in action groups and prints out error to file.
+     *
+     * @param  InputInterface $input
+     * @return void
+     * @throws Exception
+     */
+    public function execute(InputInterface $input)
+    {
+        MftfApplicationConfig::create(
+            true,
+            MftfApplicationConfig::UNIT_TEST_PHASE,
+            false,
+            MftfApplicationConfig::LEVEL_NONE,
+            true
+        );
+
+        $allModules = ModuleResolver::getInstance()->getModulesPath();
+
+        $actionGroupXmlFiles = StaticCheckHelper::buildFileList(
+            $allModules,
+            DIRECTORY_SEPARATOR . 'ActionGroup' . DIRECTORY_SEPARATOR
+        );
+
+        $this->errors = $this->findErrorsInFileSet($actionGroupXmlFiles);
+
+        $this->output = StaticCheckHelper::printErrorsToFile(
+            $this->errors,
+            self::ERROR_LOG_FILENAME,
+            self::ERROR_LOG_MESSAGE
+        );
+    }
+
+    /**
+     * Return array containing all errors found after running the execute() function.
+     * @return array
+     */
+    public function getErrors()
+    {
+        return $this->errors;
+    }
+
+    /**
+     * Return string of a short human readable result of the check. For example: "No unused arguments found."
+     * @return string
+     */
+    public function getOutput()
+    {
+        return $this->output;
+    }
+
+    /**
+     * Finds all unused arguments in given set of actionGroup files
+     * @param Finder $files
+     * @return array $testErrors
+     */
+    private function findErrorsInFileSet($files)
+    {
+        $actionGroupErrors = [];
+        foreach ($files as $filePath) {
+            $contents = file_get_contents($filePath);
+            preg_match_all(self::ACTIONGROUP_XML_REGEX_PATTERN, $contents, $actionGroups);
+            $actionGroupToArguments = $this->buildUnusedArgumentList($actionGroups[0]);
+            $actionGroupErrors += $this->setErrorOutput($actionGroupToArguments, $filePath);
+        }
+        return $actionGroupErrors;
+    }
+
+    /**
+     * Builds array of action group => unused arguments
+     * @param array $actionGroups
+     * @return array $actionGroupToArguments
+     */
+    private function buildUnusedArgumentList($actionGroups)
+    {
+        $actionGroupToArguments = [];
+
+        foreach ($actionGroups as $actionGroupXml) {
+            preg_match(self::ACTIONGROUP_NAME_REGEX_PATTERN, $actionGroupXml, $actionGroupName);
+            $unusedArguments = $this->findUnusedArguments($actionGroupXml);
+            if (!empty($unusedArguments)) {
+                $actionGroupToArguments[$actionGroupName[1]] = $unusedArguments;
```

Comment:
Same as above. Are we always going to get matches?

Reply:
since it's a multi line match pattern, even if it's a non-match the $actionGroups always returns an array with an array of size 0; So, it's a safe use of the index.